pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,644 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the clifford simulator using stim
+"""
+
+from dataclasses import replace
+from functools import partial
+from numbers import Number
+from typing import Union, Tuple, Optional, Sequence
+import concurrent.futures
+import numpy as np
+
+import pennylane as qml
+from pennylane.tape import QuantumTape, QuantumScript
+from pennylane.typing import Result, ResultBatch
+from pennylane.transforms import convert_to_numpy_parameters
+from pennylane.transforms.core import TransformProgram
+from pennylane.measurements import ExpectationMP, StateMP, DensityMatrixMP, PurityMP
+from pennylane.devices.qubit.sampling import get_num_shots_and_executions
+
+from . import Device
+from .execution_config import ExecutionConfig, DefaultExecutionConfig
+
+from .default_qubit import accepted_sample_measurement
+
+from .preprocess import (
+    decompose,
+    validate_observables,
+    validate_measurements,
+    validate_multiprocessing_workers,
+    validate_device_wires,
+    warn_about_trainable_observables,
+)
+
+Result_or_ResultBatch = Union[Result, ResultBatch]
+QuantumTape_or_Batch = Union[QuantumTape, Sequence[QuantumTape]]
+
+# Updated observable list
+_MEAS_OBSERVABLES = {
+    "PauliX",
+    "PauliY",
+    "PauliZ",
+    "Hadamard",
+    "Hermitian",
+    "Identity",
+    "Projector",
+    "Hamiltonian",
+    "Sum",
+    "SProd",
+    "Prod",
+    "Exp",
+}
+
+# Clifford gates
+_GATE_OPERATIONS = {
+    "Identity": "I",
+    "PauliX": "X",
+    "PauliY": "Y",
+    "PauliZ": "Z",
+    "Hadamard": "H",
+    "S": "S",
+    "Adjoint(S)": "S_DAG",
+    "SX": "SX",
+    "Adjoint(SX)": "SX_DAG",
+    "CNOT": "CNOT",
+    "SWAP": "SWAP",
+    "ISWAP": "ISWAP",
+    "Adjoint(ISWAP)": "ISWAP_DAG",
+    "CY": "CY",
+    "CZ": "CZ",
+    "GlobalPhase": None,
+    "BasisState": None,
+    "StatePrep": None,
+    "Snapshot": None,
+    "Barrier": None,
```

Comment:
What is the purpose of the `None`s?

Reply:
Is `_GATE_OPERATIONS` for two things:
- What is supported
- How to map to stim-specific gates?

Reply:
Yep, that's the primary motivation for the `_GATE_OPERATIONS`. The `None` values allow us to use the PL gates that we want to support, but they don't have a `stim`-specific corresponding operation.